app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java

```diff
@@ -343,10 +456,63 @@ public static Graph addNodesToGraph(String[] contents)
                 if (i == 0)
                     current = coordinatesToInsert.getHead();
                 if (current != null)
-                    returnMe.insertVertex((LatLng)current.getEle());
+                    if (i < nmbClassNodes) // i < nmbClassNodes the condition causing the error
+                    {
+                        returnMe.insertVertex((LatLng)current.getEle(), 0); //insert a class node
+                    } else {
+                        returnMe.insertVertex((LatLng)current.getEle(), 1); //insert a hall node
+                    }
                 current = current.getNext();
             }
         }
         return returnMe;
     }
+
+    public static LatLng readClassCoordinate(String currentLine)
+    {
+        String[] lineString = currentLine.split(":");
+        String floorName = lineString[0].trim(); // ex: H-801
+
+        int currentPos = lineString[1].indexOf("{");
+        int posOfHalfway = lineString[1].indexOf(",");
+        double x_coordinate = Double.parseDouble(lineString[1].substring(currentPos+1,posOfHalfway));
+        double y_coordinate = Double.parseDouble(lineString[1].substring(posOfHalfway+2, lineString[1].length()-2));
+        LatLng currentCoordinate = new LatLng(x_coordinate, y_coordinate);
+        return currentCoordinate;
+    }
+
+    public static LatLng readHallCoordinate(String currentLine)
+    {
+        int currentPos = currentLine.indexOf("{");
+        int posOfHalfway = currentLine.indexOf(",");
+        double x_coordinate = Double.parseDouble(currentLine.substring(currentPos+1,posOfHalfway));
+        double y_coordinate = Double.parseDouble(currentLine.substring(posOfHalfway+2, currentLine.length()-2));
+        LatLng currentCoordinate = new LatLng(x_coordinate, y_coordinate);
+        return currentCoordinate;
+    }
+
+    public void addAdjacentNodes()
+    {
+        // we want to insert an edge between all of the nodes that are adjacent. i.e the nodes that are closest to eachother
+        float distance = 0;
+        for (int i = 0; i < nodes.length; i++)
+        {
+            Node currentNode = nodes[i];
+            for (int j = 0; j < nodes.length; j++)
+            {
+                distance = calculateNodeDistance(currentNode.getElement(), nodes[j].getElement());
+                if (j != i && distance < 8.5)
+                    insertEdge(currentNode.getElement(), nodes[j].getElement());
+            }
+        }
+    }
+
+    // calculates the distance between 2 nodes in meters
+    public float calculateNodeDistance(LatLng node1, LatLng node2)
+    {
+        float[] distance = new float[1];
+        distanceBetween(node1.latitude, node1.longitude, node2.latitude, node2.longitude, distance);
```

Comment:
I can't seem the find where this method is. Where is distanceBetween() defined?

Reply:
distanceBetween() is an imported method from the android.location.Location class. This method calculates the difference between 2 LatLng points in meters